src/data/int/relation.lean

```diff
@@ -0,0 +1,60 @@
+/-
+Copyright (c) 2021 Floris van Doorn. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Floris van Doorn
+-/
+import data.int.order
+
+/-!
+# Relations on `ℤ`
+
+This file contains properties about relations on `ℤ` and their closure operations
+(like the transitive closure).
+
+-/
+
+open set int
+
+namespace relation
+
+/-- `(n, m)` is in the reflexive-transitive closure of `~` if `i ~ i+1` and `i+1 ~ i`
+  for all `i` between `n` and `m`. -/
+lemma refl_trans_gen_int (r : ℤ → ℤ → Prop) {n m : ℤ}
```

Comment:
All those lemmas should be about `[linear_order α] [succ_order α] [is_succ_archimedean α]`. See for example `succ.rec_linear`.